volmdlr/wires.py

```diff
@@ -861,90 +873,90 @@ class Contour(Wire):
-            primitives.extend(self.primitives[ip1 + 1:])
-            primitives.extend(self.primitives[:ip2])
-            if primitive2.start == point2:
-                pass
-            elif primitive2.end == point2:
-                primitives.append(primitive2)
-            else:
-                primitives.append(primitive2.split(point2)[0])
-        elif (ip1 == ip2 and point1.point_distance(
-                primitive1.start) < point2.point_distance(primitive1.start)):
-            if primitive1.start != point1:
-                primitive = primitive1.split(point1)[1]
-                primitive = primitive.split(point2)[0]
-                primitives.append(primitive)
-            elif primitive1.end != point2:
-                primitive = primitive1.split(point2)[0]
-                primitives.append(primitive)
-            else:
-                primitives.append(primitive2)
+    #     if ip1 < ip2:
+    #         if primitive1.start == point1:
+    #             primitives.append(primitive1)
+    #         elif primitive1.end == point1:
+    #             pass
+    #         else:
+    #             primitives.append(primitive1.split(point1)[1])
+    #         primitives.extend(self.primitives[ip1 + 1:ip2])
+    #         if primitive2.start == point2:
+    #             pass
+    #         elif primitive2.end == point2:
+    #             primitives.append(primitive2)
+    #         else:
+    #             primitives.append(primitive2.split(point2)[0])
+    #     elif ip1 > ip2 or (ip1 == ip2 and point1.point_distance(
+    #             primitive1.start) > point2.point_distance(primitive1.start)):
+    #         if primitive1.start == point1:
+    #             primitives.append(primitive1)
+    #         elif primitive1.end == point1:
+    #             pass
+    #         else:
+    #             primitives.append(primitive1.split(point1)[1])
+    #         # primitives.append(primitive1.split(point1)[1])
+    #         primitives.extend(self.primitives[ip1 + 1:])
+    #         primitives.extend(self.primitives[:ip2])
+    #         if primitive2.start == point2:
+    #             pass
+    #         elif primitive2.end == point2:
+    #             primitives.append(primitive2)
+    #         else:
+    #             primitives.append(primitive2.split(point2)[0])
+    #     elif (ip1 == ip2 and point1.point_distance(
+    #             primitive1.start) < point2.point_distance(primitive1.start)):
+    #         if primitive1.start != point1:
+    #             primitive = primitive1.split(point1)[1]
+    #             primitive = primitive.split(point2)[0]
+    #             primitives.append(primitive)
+    #         elif primitive1.end != point2:
+    #             primitive = primitive1.split(point2)[0]
+    #             primitives.append(primitive)
+    #         else:
+    #             primitives.append(primitive2)
 
-        return primitives
+    #     return primitives
```

Comment:
You can delete this commented block of code

Reply:
@WirajanDASILVA So, you agree to delete extract_primitives from Contour, and use Wire's method ?

Reply:
Yes! It seems like it works ok for all operations, since drone has not identified any issues when running the unittests. You can write a very explicit message to the commit like "remove Contour.extract_primitives", so if we have to come back to it later, we'll easily find the corresponding commit.

Reply:
It is done !